jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -74,44 +77,35 @@
   private final ScopeManager scopeManager;
   private final BaggageSetter baggageSetter;
   private final boolean expandExceptionLogs;
-
-  private JaegerTracer(
-      String serviceName,
-      Reporter reporter,
-      Sampler sampler,
-      PropagationRegistry registry,
-      Clock clock,
-      Metrics metrics,
-      Map<String, Object> tags,
-      boolean zipkinSharedRpcSpan,
-      ScopeManager scopeManager,
-      BaggageRestrictionManager baggageRestrictionManager,
-      boolean expandExceptionLogs) {
-    this.serviceName = serviceName;
-    this.reporter = reporter;
-    this.sampler = sampler;
-    this.registry = registry;
-    this.clock = clock;
-    this.metrics = metrics;
-    this.zipkinSharedRpcSpan = zipkinSharedRpcSpan;
-    this.scopeManager = scopeManager;
-    this.baggageSetter = new BaggageSetter(baggageRestrictionManager, metrics);
-    this.expandExceptionLogs = expandExceptionLogs;
+  private final JaegerObjectFactory objectFactory;
+
+  protected JaegerTracer(JaegerTracer.Builder builder) {
+    this.serviceName = builder.serviceName;
+    this.reporter = builder.reporter;
+    this.sampler = builder.sampler;
+    this.registry = builder.registry;
+    this.clock = builder.clock;
+    this.metrics = builder.metrics;
+    this.zipkinSharedRpcSpan = builder.zipkinSharedRpcSpan;
+    this.scopeManager = builder.scopeManager;
+    this.baggageSetter = new BaggageSetter(builder.baggageRestrictionManager, metrics);
+    this.expandExceptionLogs = builder.expandExceptionLogs;
+    this.objectFactory = builder.objectFactory;
 
     this.version = loadVersion();
 
-    tags.put(Constants.JAEGER_CLIENT_VERSION_TAG_KEY, this.version);
-    if (tags.get(Constants.TRACER_HOSTNAME_TAG_KEY) == null) {
+    builder.tags.put(Constants.JAEGER_CLIENT_VERSION_TAG_KEY, this.version);
```

Comment:
nit: instead of modifying `builder.tags` I would create a local variable as a copy
```java
Map<String, String> tags = new HashMap<String, String>(builder.tags);
```